iaso/api/pages.py

```diff
@@ -65,5 +64,8 @@ def get_object(self):
     def get_queryset(self):
         user = self.request.user
         order = self.request.query_params.get("order", "created_at").split(",")
-        users = User.objects.filter(iaso_profile__account=user.iaso_profile.account)
-        return Page.objects.filter(users__in=users).order_by(*order).distinct()
+        if user.has_perm(permission.PAGES) and not user.has_perm(permission.PAGE_WRITE):
+            return Page.objects.filter(users=user).order_by(*order).distinct()
+        
+        return Page.objects.filter(users__iaso_profile__account=user.iaso_profile.account).order_by(*order).distinct()
```

Comment:
It works as expected, but as a developer it's not really clear to me that the pages are filtered on the `account`. This was already unclear in the code before, and there's actually an `account_id` on the `Page` model, so what do you think about something like this?

```python
        queryset = Page.objects.filter(account=user.iaso_profile.account)

        if user.has_perm(permission.PAGES) and not user.has_perm(permission.PAGE_WRITE):
            queryset = queryset.filter(users=user)

        return queryset.order_by(*order).distinct()
```